store/sqlstore/store.go

```diff
@@ -1245,6 +1265,50 @@ func (ss *SqlStore) UpdateLicense(license *model.License) {
 	ss.license = license
 }
 
+func (ss *SqlStore) migrate(conn *sql.DB) error {
+	var driver database.Driver
+	var err error
+
+	if ss.DriverName() == model.DATABASE_DRIVER_MYSQL {
+		driver, err = mysqlmigrate.WithInstance(conn, &mysqlmigrate.Config{})
+		if err != nil {
+			return err
+		}
+	} else {
+		driver, err = postgres.WithInstance(conn, &postgres.Config{})
+		if err != nil {
+			return err
+		}
+	}
+
+	var assetNamesForDriver []string
+	for _, assetName := range migrations.AssetNames() {
+		if strings.HasPrefix(assetName, ss.DriverName()) {
+			assetNamesForDriver = append(assetNamesForDriver, strings.TrimPrefix(assetName, ss.DriverName()+"/"))
```

Comment:
We need to do this weird prefix trimming and then prefix add, because golang migrate has a regex that does not allow paths on go-bindata files

Reply:
Can you point me to which regex are we talking about? Just for my understanding. When doing this for mattermod, I did not notice any such limitation.

Reply:
I am using folder namespacing in order to separate files for MySQL and Postgresql. So the bindata files are looking something like: `mysql/123_name.up.sql` and `postgres/123_name.up.sql` the golang-migrate has a parses that expects just a file name https://github.com/golang-migrate/migrate/blob/master/source/parse.go#L21 without paths.

Reply:
Ohh got you now, so it's trimming the folder prefix. I thought it was something else. Wondering if we can use [filepath.Base](https://golang.org/pkg/path/filepath/#Base) for this? That seems like a more robust alternative.

Reply:
Will try and see if it works.